src/transformers/lib_integrations/peft/peft_mixin.py

```diff
@@ -59,14 +64,15 @@ class PeftAdapterMixin:
 
     def load_adapter(
         self,
-        peft_model_id: str,
+        peft_model_id: Union[str, Dict[str, "torch.Tensor"]],
```

Comment:
Should we choose a different name, now that a state dict can be passed? Alternatively, we could add another (optional) argument to pass the state dict, make `peft_model_id` optional, and do a check that exactly one of the two should be passed.

Reply:
Yes makes sense to me! `peft_model_id_or_state_dict` sounds great - was just wondering about breaking change here. But think this function is not yet in a release so should probs be fine to change no?